false positive no-member for static derived member

I know there are problems for dynamically generated members, but in this case everything is static.
### Steps to reproduce
execute pylint on this code:
```
class Base:
    label: str

class Derived(Base):
    label = "I exist!"

print(Derived.label)
```
### Current behavior
```
test.py:7:6: E1101: Class 'Derived' has no 'label' member (no-member)
```
reported by pylint

### Expected behavior
No `no-member` error reported by pylint

### pylint --version output
```
~:$ pylint --version
pylint 2.6.0
astroid 2.4.2
Python 3.8.2 (default, Jul 16 2020, 14:00:26) 
[GCC 9.3.0]
```
tested also on
```
~:$ pylint --version
pylint 2.4.4
astroid 2.3.3
Python 3.8.2 (default, Jul 16 2020, 14:00:26) 
[GCC 9.3.0]
```

--- a/tests/functional/n/no/no_member.py
+++ b/tests/functional/n/no/no_member.py
@@ -29,3 +29,16 @@ sorted_query = parse.urlencode(
     sorted(parse.parse_qsl(parsed_url.query), key=lambda param: param[0]))
 new_parsed_url = parse.ParseResult._replace(parsed_url, query=sorted_query)
 new_url = new_parsed_url.geturl()  # No error here
+
+
+# Regression test for https://github.com/PyCQA/pylint/issues/3803
+# pylint: disable=too-few-public-methods
+class Base:
+    label: str
+
+
+class Derived(Base):
+    label = "I exist!"
+
+
+print(Derived.label)

